ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -2999,25 +3055,46 @@ const triggerSet: TriggerSet<Data> = {
             ko: '어둠 기둥 (교체)',
           },
         };
-
-        let tower: typeof data.pangenesisCurrentColor;
-        if (data.pangenesisCurrentColor === 'light')
-          tower = 'dark';
-        else if (data.pangenesisCurrentColor === 'dark')
-          tower = 'light';
-        else
-          tower = data.lastPangenesisTowerColor;
-
-        if (tower === undefined)
+        const strat = data.triggerSetConfig.pangenesisFirstTower;
+        const myRole = data.pangenesisRole[data.me];
+        if (myRole === undefined)
           return;
 
-        const isSameTower = tower === data.lastPangenesisTowerColor;
-        if (isSameTower)
-          return { infoText: tower === 'light' ? output.lightTower!() : output.darkTower!() };
+        const switchOutput = data.lastPangenesisTowerColor === 'light'
+          ? 'darkTowerSwitch'
+          : 'lightTowerSwitch';
+        const stayOutput = data.lastPangenesisTowerColor === 'light' ? 'lightTower' : 'darkTower';
+
+        // 2nd towers
+        if (data.pangenesisTowerCount === 1) {
+          if (strat === 'not') {
+            // in the 0+2 strat, 2nd tower responsibilities are fixed based on 1st tower soaks.
+            // the shortLight, shortDark, and both 'not' players always take the northern (opposite color) towers.
+            // the longLight, longDark, and both 'one' players soak the southern (same color) towers - per their still-active Pangenesis Initial call
+            const swapRoles: PangenesisRole[] = ['not', 'shortLight', 'shortDark'];
+            if (swapRoles.includes(myRole))
+              return { infoText: output[switchOutput]!() }; // infoText because, although a switch, it's 100% anticipated and should be treated as a reminder
+            return;
```

Comment:
Should we call out the "stay output" for the other people waiting to take the second tower? I guess you can only know what the color is for the longs, though.  Hmm.

Reply:
> Should we call out the "stay output" for the other people waiting to take the second tower? I guess you can only know what the color is for the longs, though. Hmm.

I avoided doing a call here on the 2nd towers for the other people, because they still have the Pangenesis Initial call up (longer duration) that tells them which second tower to take, and doing a new infoalert saying the same thing feels noisy.

Reply:
Ok, sounds good!